core_impl/src/main/java/io/opencensus/implcore/stats/StatsManager.java

```diff
@@ -32,25 +33,34 @@
   // clock used throughout the stats implementation
   private final Clock clock;
 
+  private final CurrentStatsState state;
   private final MeasureToViewMap measureToViewMap = new MeasureToViewMap();
 
-  StatsManager(EventQueue queue, Clock clock) {
+  StatsManager(EventQueue queue, Clock clock, CurrentStatsState state) {
     checkNotNull(queue, "EventQueue");
     checkNotNull(clock, "Clock");
+    checkNotNull(state, "state");
     this.queue = queue;
     this.clock = clock;
+    this.state = state;
   }
 
   void registerView(View view) {
     measureToViewMap.registerView(view, clock);
   }
 
   ViewData getView(View.Name viewName) {
-    return measureToViewMap.getView(viewName, clock);
+    return measureToViewMap.getView(viewName, clock, state.get());
   }
 
   void record(TagContext tags, MeasureMapInternal measurementValues) {
-    queue.enqueue(new StatsEvent(this, tags, measurementValues));
+    if (state.get() == StatsCollectionState.ENABLED) {
```

Comment:
I think we could improve performance when stats is disabled by checking the state in StatsRecorderImpl.newMeasureMap and returning a no-op MeasureMap in that case.  It doesn't need to be in this PR, though.